clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/BaseKeyValueOperation.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.internal.store.operations;
+
+/**
+ * Base class that represents {@link org.ehcache.Cache} operations
+ * that involves a key and a value.
+ *
+ * @param <K> key type
+ * @param <V> value type
+ */
+public abstract class BaseKeyValueOperation<K, V> extends BaseOperation<K> implements KeyValueOperation<K, V> {
```

Comment:
Since the initial implementation of clutered-tier only cache is not gonna use any of the metadata we'll keep the codecs handle keys and values only. This change can be done once we start using the metadata.